hazelcast/src/main/java/com/hazelcast/collection/impl/set/SetContainer.java

```diff
@@ -76,6 +76,13 @@ public SetConfig getConfig() {
         if (itemSet == null) {
             if (itemMap != null && !itemMap.isEmpty()) {
                 itemSet = new HashSet<CollectionItem>(itemMap.values());
+                long maxItemId = Long.MIN_VALUE;
```

Comment:
This is a fix comparable to https://github.com/hazelcast/hazelcast/pull/2844 and https://github.com/hazelcast/hazelcast/pull/4275. I don't like the double iteration of the items not much, but the `new HashSet<CollectionItem>(itemMap.values());` has optimized values for the initial size, which I don't like to rip out of the JDK.

Any good ideas how to optimize this? Or should we leave it like this, since this only happens when a backup owner gets promoted to be the new primary owner. So it's quite rare.